internal/internal_nexus_task_handler.go

```diff
@@ -416,3 +422,89 @@ func (p *payloadSerializer) Serialize(v any) (*nexus.Content, error) {
 }
 
 var emptyReaderNopCloser = io.NopCloser(bytes.NewReader([]byte{}))
+
+// convertServiceError converts a serviceerror into a Nexus HandlerError if possible.
+// If exposeDetails is true, the error message from the given error is exposed in the converted HandlerError, otherwise,
+// a default message with minimal information is attached to the returned error.
+// Roughly taken from https://github.com/googleapis/googleapis/blob/master/google/rpc/code.proto
+// and
+// https://github.com/grpc-ecosystem/grpc-gateway/blob/a7cf811e6ffabeaddcfb4ff65602c12671ff326e/runtime/errors.go#L56.
+func convertServiceError(err error, exposeDetails bool) error {
```

Comment:
Doesn't seem like there's much value in splitting `convertServiceError` and `convertApplicationError`, might as well just have `convertError`. Also doesn't seem like `exposeDetails` parameter has any value since it is never set to false on this private function. Can wait for the future need to hide details before adding that functionality, it's dead currently.

Reply:
Agree there's not much value, for the split. Initially there was more logic in the application error conversion since I thought we may want to handle other TemporalErrors but I can merge.

I already know that we'll want to allow hiding details and had this logic in the server codebase (where I ported this status error conversion logic from) so I figured I'd keep it in here.

Reply:
Restructured slightly.